ci/travis/ci.sh

```diff
@@ -4,7 +4,7 @@
 
 set -eo pipefail && if [ -n "${OSTYPE##darwin*}" ]; then set -ux; fi  # some options interfere with Travis's RVM on Mac
 
-ROOT_DIR="$(cd "$(dirname "${BASH_SOURCE:-$0}")"; pwd)"
+ROOT_DIR="$(builtin cd "$(dirname "${BASH_SOURCE:-$0}")"; pwd)"
```

Comment:
This is not maintainable. Developers can't remember to always use `builtin cd` for this when writing CI script. Is there another way to do this? What's the RVM issue?

Reply:
There are a couple issues to deal with:

- Travis on Mac at least seems to use RVM (a Ruby thing?) to hook `cd` for whatever reason, so whenever you run `cd` with `set -x`, it floods the screen with a ton of useless messages. `builtin cd` bypasses that.

- Some of those scripts also can't function with `set -u`; they error out and cause the scripts to fail, so we have to restore the `set -u` state before returning to the caller.

Forgetting `builtin` is already not a _big_ deal, since it merely floods the screen. The part about `set -u` is harder to get around though. I'll continue on that below. (...)